src/main/java/org/codehaus/plexus/archiver/xz/PlexusIoXZResourceCollection.java

```diff
@@ -47,17 +50,7 @@ protected String getDefaultExtension()
     @Override
     protected InputStream getInputStream( File file ) throws IOException
     {
-        FileInputStream fileIs = new FileInputStream( file );
-        try
-        {
-            final InputStream result = XZUnArchiver.getXZInputStream( fileIs );
-            fileIs = null;
-            return result;
-        }
-        finally
-        {
-            IOUtil.close( fileIs );
-        }
+        return XZUnArchiver.getXZInputStream( Streams.fileInputStream( file ) );
```

Comment:
previous code was closing the created InputStream but new one not anymore.
Is it correct?

Reply:
Yes, it is totally correct, the InputStream was actually never closed, the reference was set to `null` and `IOUtil.close(fis)` don't close on null reference, yet even if when the reference was set to null `getXZInputStream` acts as a decorator and was holding the reference of the InputStream that was later returned.

Actually trying to close the InputStream would be a bug since it's used in the returned decorator and if was closed when consumed it would throw a `java.io.IOException: Stream Closed` or a `java.nio.channels.ClosedChannelException`